File: AULA_08/JAVA/GeraPalindromo.java
/* GeraPalindromo.java
 * Crie uma rotina que retorne o inverso de um número
 * Use tal rotina para gerar um palindromo.
 *
 * Entrada: n (int) 
 * Saída: palindromo
 *
 * Autor: Fabrício Olivetti de França
 * Disciplina Processamento da Informação
 * Universidade Federal do ABC
 */


import java.util.Scanner;

class GeraPalindromo{

	// subrotina para  gerar o inverso de um número
	public static int Inverso( int n ){
		int inverso=0;
		while( n!=0 ){
			inverso = inverso*10 + (n%10);
			n /= 10;
		}
		return inverso;
	}

	// principal
	public static void main(String[] args) {
		Scanner leitor = new Scanner(System.in);
		int n;
		n = leitor.nextInt();
		int maxTentativas = 100;
		int i=0;
		int inv=0;
		for( ;i<maxTentativas;i++ ){
			inv = Inverso(n); // inverte o número
			if( n == inv ){ // se for palíndromo, beleza
				break;
			}
			n += inv; // senão soma n + inverso
		}
		if( inv == n ){
			System.out.println("Obtive "+n+ " após "+i+" tentativas");
		}else{
			System.out.println("Desisto");
		}

	}
}

File: AULA_08/Python/GeraPalindromo.py
# -*- coding: cp1252 -*-
'''
GeraPalindromo.py
 * Crie uma rotina que retorne o inverso de um número
 * Use tal rotina para gerar um palindromo.
 *
 * Entrada: n (int) 
 * Saída: palindromo
 *
 * Autor: Fabrício Olivetti de França
 * Disciplina Processamento da Informação
 * Universidade Federal do ABC
'''

def Inverso( n ):
    inverso = 0
    while n!= 0:
        inverso = inverso*10 + n%10
        n /= 10
    return inverso

if __name__ == "__main__":

    n = int(raw_input("Digite um número: "))
    maxTentativas = 100
    inv = 0
    for it in range(maxTentativas):
        inv = Inverso(n)
        if n==inv:
            break
        n += inv
    if inv==n:
        print "Obtive ",n," após",it,"tentativas"
    else:
        print "Desisto"
